lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -61,7 +61,36 @@ static void preFileOpen(void*)
 {
   TF_DEBUG(ALUSDMAYA_EVENTS).Msg("preFileOpen\n");
 }
+//----------------------------------------------------------------------------------------------------------------------
+static void initialiseChildAndSublayers(nodes::Layer& layer, nodes::ProxyShape* proxy)
+{
+  // initialise self
+  MPlug plug = layer.nameOnLoadPlug();
+  MString path = plug.asString();
+  if(path.length() && path.substring(0, 3) != "anon")
```

Comment:
I noticed we had this substring check before. Maybe we can do it better this time, try use SdfLayer::IsAnonymousLayerIdentifier to determine if it's anonymous